lib/mix/tasks/nettools.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Mix.Tasks.Archethic.Nettools do
+  @moduledoc """
+  Provides Support for Networking tools At cli
+
+  ## Command line options
+
+    * `-h`, `--help`  - show this help
+    * `-p`, `--punch` - Attempt to open ports
+    * `-i`, `--ip`    - Public IP
+
+  ## Command line arguments
+    * `punch` - args: []-> 30002 40002, list_of ports
+
+  ## Example
+
+  ```sh
+  mix archethic.nettools -p
+  mix archethic.nettools --punch
+  mix archethic.nettools -p 30_000 40_000
+  mix archethic.nettools -i
+  mix archethic.nettools --ip
+  ```
+
+  """
+  use Mix.Task
+  alias Archethic.Networking
+
+  require Logger
+
+  @impl Mix.Task
+  def run(args) do
+    args
+    |> parse_args
+  end
+
+  def parse_args(args) do
+    OptionParser.parse(args,
+      strict: [help: :boolean, punch: :boolean, ip: :boolean],
+      aliases: [h: :help, p: :punch, i: :ip]
+    )
+    |> args_to_internal_representation()
+  end
+
+  def args_to_internal_representation({[help: true], [], []}) do
+    Mix.shell().cmd("mix help #{Mix.Task.task_name(__MODULE__)}")
+  end
+
+  def args_to_internal_representation({[punch: true], _port_list = [port | more_ports], errors}) do
+    Networking.PortForwarding.try_open_port(abs(Integer.parse(port) |> elem(0)), true)
+    args_to_internal_representation({[punch: true], more_ports, errors})
+  end
+
+  def args_to_internal_representation({[punch: true], _port_list = [], _errors}) do
+    Networking.PortForwarding.try_open_port(40_000, true)
+    Networking.PortForwarding.try_open_port(30_002, true)
+  end
+
+  def args_to_internal_representation({[ip: true], _, _}) do
+    {_, ip} = Networking.IPLookup.RemoteDiscovery.get_node_ip()
```

Comment:
I suggest you to use Networking.IPLookup.get_node_ip to leverage fallback for NAT discovery as well, 

you will also need to setup the default  IPLookup as in dev, it's Static

```elixir
Application.put_env(:archethic, Archethic.Networking.IPLookup,
Archethic.Networking.IPLookup.NATDiscovery
```